lib/valueflow.cpp

```diff
@@ -1658,8 +1658,79 @@ static bool evalAssignment(ValueFlow::Value &lhsValue, const std::string &assign
     return true;
 }
 
+template<class T>
+struct SingleRange {
+    T* x;
+    T* begin() const {
+        return x;
+    }
+    T* end() const {
+        return x+1;
+    }
+};
+
+template<class T>
+SingleRange<T> MakeSingleRange(T& x)
+{
+    return {&x};
+}
+
+class MultiRange {
```

Comment:
This is rather generic name for a type that isn't generic at all. It really should be something like `SelectVariableFromVarIdMapRange`.

Reply:
done